views/Data.js

```diff
@@ -422,6 +423,260 @@ export function useSetFlowToBasedOnData({
+    context,
+    path,
+    validate,
+    viewPath,
+  })
+  let dataIsValid = useDataIsValid({
+    context,
+    path,
+    validate,
+    viewPath,
+  })
+
+  let setFlowTo = useSetFlowTo(viewPath)
+  let checkInitialRef = useRef(false)
+  let renderOneRef = useRef(false)
+
+  useDataListener({
+    context,
+    viewPath,
+    listener: (next, prev, flow, setFlowTo) => {
+      if (!renderOneRef.current) return
+
+      let value = path ? get(next, path) : next
+      let isContent = validate(value, value, next)
+
+      let prevValue = path ? get(prev, path) : prev
+      let isContentPrev = validate(prevValue, prevValue, prev)
+
+      let target = normalizePath(viewPath, isContent ? isValid : no)
+
+      if (
+        isContent !== isContentPrev ||
+        checkInitialRef.current ||
+        !flow.has(target)
+      ) {
+        setFlowTo(target)
+      }
+
+      if (checkInitialRef.current) {
+        checkInitialRef.current = false
+      }
+    },
+  })
+
+  useEffect(() => {
+    // on mount
+    let isContent = checkInitial ? dataIsValidInitial : dataIsValid
+    let target = normalizePath(viewPath, isContent ? isValid : no)
+    setFlowTo(target)
+    checkInitialRef.current = !checkInitial
+    renderOneRef.current = true
+  }, [viewPath]) //eslint-disable-line
+
+  return null
+}
+
+export function useSetFlowToBasedOnDataIsInvalid({
+  context,
+  path,
+  viewPath,
+  validate,
+  checkInitial = true,
+  isValid = 'Content',
+  no = 'No',
+}) {
+  let dataIsValidInitial = useDataIsValidInitial({
+    context,
+    path,
+    validate,
+    viewPath,
+  })
+  let dataIsValid = useDataIsValid({
+    context,
+    path,
+    validate,
+    viewPath,
+  })
+
+  let setFlowTo = useSetFlowTo(viewPath)
+  let checkInitialRef = useRef(false)
+  let renderOneRef = useRef(false)
+
+  useDataListener({
+    context,
+    viewPath,
+    listener: (next, prev, flow, setFlowTo) => {
+      if (!renderOneRef.current) return
+
+      let value = path ? get(next, path) : next
+      let isContent = !validate(value, value, next)
+
+      let prevValue = path ? get(prev, path) : prev
+      let isContentPrev = !validate(prevValue, prevValue, prev)
+
+      let target = normalizePath(viewPath, isContent ? isValid : no)
+
+      if (
+        isContent !== isContentPrev ||
+        checkInitialRef.current ||
+        !flow.has(target)
+      ) {
+        setFlowTo(target)
+      }
+
+      if (checkInitialRef.current) {
+        checkInitialRef.current = false
+      }
+    },
+  })
+
+  useEffect(() => {
+    // on mount
+    let isContent = checkInitial ? !dataIsValidInitial : !dataIsValid
+    let target = normalizePath(viewPath, isContent ? isValid : no)
+    setFlowTo(target)
+    checkInitialRef.current = !checkInitial
+    renderOneRef.current = true
+  }, [viewPath]) //eslint-disable-line
+
+  return null
+}
+
+export function useSetFlowToBasedOnDataIsSubmitting({
+  context,
+  viewPath,
+  isSubmitting = 'IsSubmitting',
```

Comment:
shouldn't we use the same pattern here, eg `IsSubmitting` for the content view and in `IsValid` etc? (it reads `Content` in there)

Reply:
also, can the names of those views be configurable? how would the pattern be matched by the morpher otherwise? or is your idea that this choice is available if you use it on a logic file for example?